intake/utils.py

```diff
@@ -112,7 +112,7 @@ def __getstate__(self):
         # OrderedDict steers dask toward a different and faster tokenization.
         kwargs = collections.OrderedDict({k: arg.__getstate__()
                   if isinstance(arg, DictSerialiseMixin) else arg
-                  for k, arg in self._captured_init_kwargs.items()})
+                  for k, arg in sorted(self._captured_init_kwargs.items(), key=lambda x: x[0])})
```

Comment:
I do not think we should sort here.  Python guarantees that kwarg order is maintained so there can be meaningful information in that order that we will be losing by sorting.

Reply:
I think this is here because in older pythons, the dict order was *not* fixed, so sources that ought to have been equal turned out not to be.

Reply:
Isn't it better to have some cache misses than to have erroneous cache collisions? If we want to an optimized path for Python 3.5 and lower, we can sort conditional on `sys.version_info < (3, 6)`?

Reply:
Certainly agree that code shouldn't change just for the sake of what is testable...